core/src/main/scala/org/allenai/common/Version.scala

```diff
@@ -62,7 +62,8 @@ object GitVersion {
   */
 case class Version(
     git: GitVersion,
-    artifactVersion: String
+    artifactVersion: String,
+    cacheKey: Option[String] = None
```

Comment:
Add documentation for this. Also, I'd avoid having a default here - `Version` should be created through one of the local helpers.